aleph/authz.py

```diff
@@ -156,24 +158,49 @@ def from_role(cls, role):
         return cls(role.id, roles, is_admin=role.is_admin, is_blocked=role.is_blocked)
 
     @classmethod
-    def from_token(cls, token, scope=None):
-        if token is None:
-            return
-        try:
-            data = jwt.decode(token, key=settings.SECRET_KEY, verify=True)
-            if "s" in data and data.get("s") != scope:
-                raise Unauthorized()
-            return cls(
-                data.get("u"), data.get("r"), data.get("a", False), data.get("b", False)
-            )
-        except (jwt.DecodeError, TypeError):
-            return
-
-    @classmethod
-    def flush(cls):
+    def flush(cls) -> None:
         cache.kv.delete(cls.PREFIX)
 
     @classmethod
-    def flush_role(cls, role_id):
-        keys = [cache.key(a, role_id) for a in (cls.READ, cls.WRITE)]
+    def flush_role(cls, role_id) -> None:
+        keys = [cache.key(a, role_id) for a in ActionEnum]
         cache.kv.hdel(cls.PREFIX, *keys)
+
+
+class InvalidJwtToken(Exception):
+    pass
+
+
+class _JtwToken(pydantic.BaseModel):
+    u: int  # role ID
+    r: List[int]  # role IDs  # TODO(AD): Is this required or optional?
```

Comment:
I wasn't sure what this field was supposed to contain or what it meant but I haven't changed the behavior.